config/configcompression/compressiontype.go

```diff
@@ -3,30 +3,43 @@
 
 package configcompression // import "go.opentelemetry.io/collector/config/configcompression"
 
-import "fmt"
+import (
+	"fmt"
+
+	"github.com/klauspost/compress/zlib"
+)
 
 // Type represents a compression method
 type Type string
 
+type Level int
+
+type CompressionConfig struct {
+	Level Level `mapstructure:"level"`
+}
+
 const (
-	TypeGzip    Type = "gzip"
-	TypeZlib    Type = "zlib"
-	TypeDeflate Type = "deflate"
-	TypeSnappy  Type = "snappy"
-	TypeZstd    Type = "zstd"
-	TypeLz4     Type = "lz4"
-	typeNone    Type = "none"
-	typeEmpty   Type = ""
+	TypeGzip    Type  = "gzip"
+	TypeZlib    Type  = "zlib"
+	TypeDeflate Type  = "deflate"
+	TypeSnappy  Type  = "snappy"
+	TypeZstd    Type  = "zstd"
+	TypeLz4     Type  = "lz4"
+	typeNone    Type  = "none"
+	typeEmpty   Type  = ""
+	LevelNone   Level = 0
 )
 
+var typ Type
```

Comment:
We cannot use global variable for this. It's not guaranteed that `UnmarshalText` is called before `Validate` and is part of the same parent object

Reply:
We need to have the `Validate` func at a parent struct. I see two options:

1. We can wrap Type and CompressionConfig in one struct that will be squashed in the `confighttp`. Then we can have `Validate` with access to both fields. It'll be a breaking change for Go API but not for users.

2. Do the validation on `confighttp.ClientConfig`. The problem with that is that we couple `confighttp` and `configcompression` even more. It may be wort just removing `configcompression` in that case...

Reply:
yeah, that global variable was hacky. If we want to stay away from any breaking change, option 2 might be the way.

Reply:
I chose option2 and moved the validation to ClientConfig. If we want to avoid API breaking changes, we could keep the `configcompression` package.

Reply:
> If we want to avoid API breaking changes, we could keep the configcompression package.
Right. We don't need to do anything with that right now. I was thinking long term